addon/components/x-select.js

```diff
@@ -200,6 +200,10 @@ export default Ember.Component.extend({
    */
   unregisterOption: function(option) {
     this.get('options').removeObject(option);
-    this._updateValueSingle();
+
+    // We don't want to update the value if we're tearing the component down.
+    if (!this.get('isDestroying')) {
+      this._updateValueSingle();
```

Comment:
@Robdel12: Yeah you should check that! Just wanted to add a PR to fix the `multiple=true` case. Without the fix the value will change from an array to one single value, as soon as the component tears down.

Btw just use `_setDefaultValues()` instead of `this._updateValueSingle()`, no need to write duplicate code.